woocommerce/woocommerce
# Remove usages of the AccessiblePrivateMethods trait

### Changes proposed in this Pull Request:

Deprecates the usage of the `AccessiblePrivateMethods` trait, which was introduced as a means to keep methods that exist solely as the destination of a hook as `private`, thus keeping them out of the backwards compatibility pipeline. To simplify the codebase now the preferred approach is to make the methods `public` but adding an `@internal` annotation to their documentation comment. This pull request removes all the trait usages and turns all the affected methods into `public`+`@internal`.

The trait itself will be removed in WooCommerce 10.5 so that there's enough time to refactor any possible extensions using it (which _theorically_ shouldn't be the case since the trait is in the `Internal` namespace).

The `README.md` files in `includes` and `src` are updated too to explicitly mention the meaning of the `@internal` annotation.

### How to test the changes in this Pull Request:

The gist of the review should be ensuring that for all the class that had an `use AccessiblePrivateMethod` statement:

1. The said `use` statement has been removed.
2. The `self::add_action` and `self::add_filter` statements have been converted to plain `add_action` and `add_filter`.
3. The methods that were previously the target of such `self::add_action` or `self::add_filter` and were private are now public and have got an `@internal` annotation.
4. The same for the few instances of `self::mark_method_as_accessible` and `self::mark_static_method_as_accessible` that were present (these methods aren't called, and the target methods are now `public`+`@internal`).

Note that there were a few instances of `self::add_action` and `self::add_filter` where the target method was already public, or they have a target closure instead of a target method. In these cases removing the `self::` part is enough.

Other than that, do a general smoketesting of WooCommerce to ensure that critical flows aren't broken. Test also the `wp wc palt` commands in CLI since the `Automattic\WooCommerce\Internal\ProductAttributesLookup\CLIRunner` has got its `init` method removed.

Finally, check the content added to the modified `README.md` files.

### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
